src/Framework/BuildEventArgs.cs

```diff
@@ -311,7 +312,8 @@ internal virtual void CreateFromStream(BinaryReader reader, int version)
                 {
                     int submissionId = reader.ReadInt32();
                     int projectInstanceId = reader.ReadInt32();
-                    buildEventContext = new BuildEventContext(submissionId, nodeId, projectInstanceId, projectContextId, targetId, taskId);
+                    int evaluationId = reader.ReadInt32();
```

Comment:
For back compat, you should bump up the format version number, and read the new value only when the format version is greater than or equal the new version value.

Reply:
This is binary serialization which looks like the version number is derived from the CLR version. It looks like I shouldn't mess with that?

Reply:
You're right. I had confused this with the binary logging serialization, but that one's not affected by this PR.